### ISSUE TITLE
Any way to disable probing GitHub?

### ISSUE THREAD

**Issue by grepwood** [getsops/sops#1665]
I'm using sops in a network environment where I will never ever in a million years be able to contact github.com directly. I'd like to be able to call `sops -v` without it complaining that it can't check what's the latest version.

Is there any environment variable perhaps that controls this behavior?

**Comment by grepwood** [comment]
Alright I found myself aliasing `sops` to `sops --disable-version-check`.

**Comment by felixfontein** [comment]
Yes, if you don't want `sops -v` to contact GitHub, you need to pass `--disable-version-check`.

**Comment by grepwood** [comment]
What kind of REST API call is actually used to get this info? Perhaps I could emulate it from my side of the intranet

**Comment by felixfontein** [comment]
It tries two approaches (see https://github.com/getsops/sops/blob/main/version/version.go#L136):
- First it tries https://github.com/getsops/sops/releases/latest to see where this redirects to;
- Then it tries the GitHub REST API.

**Comment by c33s** [comment]
can we please have an environment variable for that? quite uncool that a security tool leaks my ip address without consent.

**Comment by nicklasfrahm** [comment]
@c33s What should the behaviour of that environment variable be? Would you like it to be an alternative to the `--disable-version-check` flag?

**Comment by c33s** [comment]
@nicklasfrahm yes #1684 is the behavior i was looking for.

still i would vote for opt-in instead of opt-out (eu gdpr): don't contact anything until i explicitly pass a `--do-version-check` flag or set the variable `SOPS_ENABLE_VERSION_CHECK` but i assume this would a different issue, would it?

**Comment by Chaostheorie** [comment]
An opt-in would be significantly better, until finding this issue by accident I didn't know the sops CLI contacted GitHub regularly.

An alternative to an opt-in via a flag would be to ask the user on the first invocation of sops if they want to opt in, then store the result in a file in `XDG_CONFIG_HOME` (or equivalent on Windows/Mac). Another subcommand for letting a user conveniently update the stored value to opt in/out after the first invocation would likely also be a good idea for such an implementation.

**Comment by felixfontein** [comment]
3.10.0 deprecates the current behavior of checking for updates by default (#1816), it will eventually get removed. Not sure when though.
